pages/client-libraries/rust.mdx

```diff
@@ -0,0 +1,714 @@
+let _columns = connection.execute(_read_node, None);
+while let Ok(result) =  connection.fetchall() {
+    for record in result {
+        for value in record.values {
+            match value {
+                Value::Node(node) => println!("Node: {}", node),
+                value => println!("Value: {}", value),
+            }
+        }
+    }
+}
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+In this example, the match statement is used to distinguish between different types of values that can be returned from the database. In this case the query will return a node, so the rest of logic could be based on that information. 
+
+
+#### Running a queries with property map 
+
+If you want to pass a property map to the query, you can do it like this:
+
+```rust
+let _create_node = "CREATE (n:Technology {name: $name, description: $description}) RETURN n";
+let mut params = HashMap::new();
+params.insert("name".to_string(), QueryParam::String("Memgraph".to_string()));
+params.insert("description".to_string(), QueryParam::String("Fastest graph DB in the world!".to_string()));
+let _columns = connection.execute(_create_node, Some(&params));
+while let Ok(result) =  connection.fetchall() {
+    for record in result {
+        for value in record.values {
+            match value {
+                Value::Node(node) => println!("Node: {}", node),
+                value => println!("Value: {}", value),
+            }
+        }
+    }
+}
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+Using this approach, the queries will not contain hard-coded values, they can be more dynamic.
+
+
+### Process the results
+
+In order to read results and serve them back to you Rust application you need to handle them. Rust is statically typed language and proper types need to be handled. 
+Depending on the type of the request you made, you can get different results back. Lets go over a few basic examples on how to handle different types and access properties of the returned results.
+
+#### Process the node results
+
+In order to process the result you need to read them first, you can do that by running the following query:
+
+```rust
+let _read_node = "MATCH (n:Technology {name: 'Memgraph'}) RETURN n";
+let _columns = connection.execute(_read_node, None);
+while let Ok(result) =  connection.fetchall() {
+    for record in result {
+        for value in record.values {
+            match value {
+                Value::Node(node) => println!("Node: {}", node),
+                value => println!("Value: {}", value),
+            }
+        }
+    }
+}
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+Since value can be Node, Relationship, Path, or any other type, we need to match the value to the appropriate type that we expect. In this case, the value is a node, so we can access the properties of the node, if node comes along. 
```

Comment:
```suggestion
Since the value of returned results can be `Node`, `Relationship`, `Path`, or some other type, we must match the value to the appropriate type. In this case, the value is a node so we can access the node's properties if a node comes along. 
```

What does it mean: If a node come along?

Reply:
I have remove that part.